lib/compress.js

```diff
@@ -148,10 +148,19 @@ merge(Compressor.prototype, {
             node.process_expression(true);
         }
         var passes = +this.options.passes || 1;
-        for (var pass = 0; pass < passes && pass < 3; ++pass) {
+        var last_count;
+        for (var pass = 0; pass < passes; pass++) {
             if (pass > 0 || this.option("reduce_vars"))
                 node.reset_opt_flags(this, true);
             node = node.transform(this);
+            if (passes > 1) {
+                var count = 0;
+                node.walk(new TreeWalker(function() {
+                    count++;
+                }));
+                if (last_count == count) break;
```

Comment:
Must use an inequality here in case compress transforms flip between two forms and never reach steady state (as has happened in the past):
```
                if (count >= last_count) break;
```